backend/oasst_backend/utils/database_utils.py

```diff
@@ -0,0 +1,79 @@
+from functools import wraps
+
+from loguru import logger
+from sqlalchemy.exc import OperationalError
+
+
+def manage_class_db_transaction(flag=True, num_retries=3):
+    def decorator(f):
+        @wraps(f)
+        def wrapped_f(self, *args, **kwargs):
+            try:
+                for i in range(num_retries):
+                    try:
+                        result = f(self, *args, **kwargs)
+                        if flag:
+                            self.db.commit()
+                        else:
+                            self.db.flush()
+                        return result
+                    except OperationalError:
+                        self.db.rollback()
+                        pass
+                logger.exception("DATABASE_MAX_RETIRES_EXHAUSTED")
```

Comment:
I think logger.exception should be used in except handlers to log the exception .. to signal an error `logger.error()` is better

Reply:
changed this